src/platform/imx8ulp/include/platform/drivers/interrupt.h

```diff
@@ -0,0 +1,40 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright 2021 NXP
+ *
+ * Author: Zhang Peng <peng.zhang_8@nxp.com>
+ */
+
+#ifdef __SOF_DRIVERS_INTERRUPT_H__
+
+#ifndef __PLATFORM_DRIVERS_INTERRUPT_H__
+#define __PLATFORM_DRIVERS_INTERRUPT_H__
+
+#include <sof/bit.h>
+
+
+/* IRQ numbers */
+#define IRQ_DSP_SYS_NMI      0
+#define IRQ_NUM_SOFTWARE0    1
+#define IRQ_NUM_TIMER0       2
+#define IRQ_NUM_TIMER1       3
+#define IRQ_NUM_MU2          14
+#define IRQ_NUM_MU3          15
+#define IRQ_SAI5_NUM         23
+#define IRQ_SAI6_NUM         24
+#define IRQ_SAI7_NUM         25
+
+#define PLATFORM_IRQ_HW_NUM	XCHAL_NUM_INTERRUPTS
+#define PLATFORM_IRQ_CHILDREN	0 /* Each cascaded struct covers 64 IRQs */
+/* IMX: Covered steer IRQs are modulo-64 aligned. */
+#define PLATFORM_IRQ_FIRST_CHILD  0
+
+/* no irqstr in 8ulp */
+int irqstr_get_sof_int(int irqstr_int);
```

Comment:
Don't mention irqstr (from irq steer) anymore.
Rename this function, see comment from [here](https://github.com/thesofproject/sof/pull/4550/commits/a89c6034138cfaf7ffc9866ef41a7ba272bbec02#diff-8800b4fb804080ade4c5978e9462553c70928737bb61b1bc9a9cd2f39614d22dR27).

Reply:
Afraid we can't. The function irqstr_get_sof_int is called in esai.c, if we rename this function in imx8ulp platform, there will have a compiler error.

Reply:
@TangleZ just add a comment like this: ``FIXME: no irqstr in 8ulp but we must implement this function because it is requred by ESAI */